guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/clone/CloneRepositoryViewImpl.java

```diff
@@ -0,0 +1,322 @@
+import com.google.gwt.user.client.Window;
+import com.google.gwt.user.client.ui.InlineHTML;
+import com.google.gwt.user.client.ui.PopupPanel;
+import com.google.gwt.user.client.ui.Widget;
+import org.guvnor.structure.client.resources.i18n.CommonConstants;
+import org.uberfire.ext.widgets.common.client.common.BusyPopup;
+import org.uberfire.ext.widgets.common.client.common.popups.BaseModal;
+import org.uberfire.ext.widgets.common.client.common.popups.errors.ErrorPopup;
+import org.uberfire.ext.widgets.core.client.resources.i18n.CoreConstants;
+
+@Dependent
+public class CloneRepositoryViewImpl
+        extends PopupPanel implements CloneRepositoryView {
+
+    interface CloneRepositoryFormBinder
+            extends
+            UiBinder<Widget, CloneRepositoryViewImpl> {
+
+    }
+
+    private Presenter presenter;
+
+    private static CloneRepositoryFormBinder uiBinder = GWT.create( CloneRepositoryFormBinder.class );
+
+    @UiField
+    Button clone;
+
+    @UiField
+    Button cancel;
+
+    @UiField
+    ControlGroup organizationalUnitGroup;
+
+    @UiField
+    ListBox organizationalUnitDropdown;
+
+    @UiField
+    HelpInline organizationalUnitHelpInline;
+
+    @UiField
+    ControlGroup nameGroup;
+
+    @UiField
+    TextBox nameTextBox;
+
+    @UiField
+    HelpInline nameHelpInline;
+
+    @UiField
+    ControlGroup urlGroup;
+
+    @UiField
+    TextBox gitURLTextBox;
+
+    @UiField
+    HelpInline urlHelpInline;
+
+    @UiField
+    TextBox usernameTextBox;
+
+    @UiField
+    PasswordTextBox passwordTextBox;
+
+    @UiField
+    BaseModal popup;
+
+    @UiField
+    InlineHTML isOUMandatory;
+
+    @UiHandler("clone")
+    public void onCloneClick( final ClickEvent e ) {
+        presenter.handleCloneClick();
+    }
+
+    @UiHandler("cancel")
+    public void onCancelClick( final ClickEvent e ) {
+        presenter.handleCancelClick();
+    }
+
+    @Override
+    public void init(Presenter presenter, boolean isOuMandatory) {
+        this.presenter = presenter;
+
+        setWidget(uiBinder.createAndBindUi(this));
+
+        if ( !isOuMandatory ) {
+            isOUMandatory.removeFromParent();
+        }
+
+        popup.setDynamicSafe( true );
+        nameTextBox.addKeyPressHandler(new KeyPressHandler() {
+            @Override
+            public void onKeyPress(final KeyPressEvent event) {
+                nameGroup.setType(ControlGroupType.NONE);
+                nameHelpInline.setText("");
+            }
+        });
+        gitURLTextBox.addKeyPressHandler(new KeyPressHandler() {
+            @Override
+            public void onKeyPress(final KeyPressEvent event) {
+                urlGroup.setType(ControlGroupType.NONE);
+                urlHelpInline.setText("");
+            }
+        });
+    }
+
+    @Override
+    public void hideForm() {
+        BusyPopup.close();
+        popup.hide();
+        super.hide();
+    }
+
+    @Override
+    public void showForm() {
```

Comment:
No need to do this if you extend ```BaseModal``` (just change the View interface method name to ```show()```).